### PR TITLE
102 watcher refactor tests

### PR DESCRIPTION
**Pull request by yahgwai**

### CODE REVIEW

File: src/service.ts [hunk 1]
```diff
@@ -1,16 +1,17 @@
 import express, { Response } from "express";
 import httpContext from "express-http-context";
 import logger from "./logger";
-import { PublicInspectionError, PublicDataValidationError, IEthereumResponse } from "./dataEntities";
+import { PublicInspectionError, PublicDataValidationError, ApplicationError } from "./dataEntities";
 import { Raiden, Kitsune } from "./integrations";
-import { Watcher } from "./watcher";
+import { Watcher } from "./watcher/watcher";
```
Comment (bigspider) [inline src/service.ts]: Perhaps I would add a ./watcher/index.ts to simplify imports.

File: src/watcher/appointmentSubscriber.ts [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    (...args: any[]): void;
+    appointmentId: string;
+}
+
+/**
+ * If we wish to unsubscribe an appointment from the provider we have a problem. In the provider
+ * appointments are keyed only by their filter, so an appointment could be updated, then another
+ * process tries to remove the old appointment and in doing so removes the new one. This class
+ * assigns appointment ids to listeners and ensures that appointments are only removed from the
+ * provided if the provide appointment id matches.
```
Comment (bigspider) [inline src/watcher/appointmentSubscriber.ts]: Typo in the last sentence of the comment.

File: src/watcher/garbageCollector.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        logger.info(a.formatLog(`GC: Appointment with end: ${a.endTime}.`))
+                    );
+                }
+
+                this.collecting = false;
+            }
+        } catch (doh) {
+            // errors escaping this block will cause unhandled promise rejections so we log and swallow here
+            logger.error("GC: Unexpected error.");
+            logger.error(doh);
```
Comment (bigspider) [inline src/watcher/garbageCollector.ts]: I think the lock should released outside of the try/catch, or it will be permanently locked in case of error.

Comment (yahgwai) [reply src/watcher/garbageCollector.ts]: good catch

File: src/service.ts (hunk 1 above)
Comment (yahgwai) [reply src/service.ts]: will do

File: src/watcher/store.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                delete this.appointmentsByStateLocator[appointmentById.getStateLocator()];
+            }
+            return true;
+        }
+        return false;
+    }
+
+    async getExpiredSince(expiryTime: number): Promise<IEthereumAppointment[]> {
+        return Object.keys(this.appointmentsById)
+            .map(a => this.appointmentsById[a])
```
Comment (bigspider) [inline src/watcher/store.ts]: Aren't the last two lines equivalent to ```Object.values(this.appointmentsById)```?

File: src/watcher/appointmentSubscriber.ts (hunk 2 above)
Comment (yahgwai) [reply src/watcher/appointmentSubscriber.ts]: fixed

File: src/watcher/store.ts (hunk 4 above)
Comment (yahgwai) [reply src/watcher/store.ts]: Yeah. It seems we needed to switch tscongif to use the es2017 before we could use. Done that now.

**Pull request merged**